junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/DiscoveryIssueCollector.java

```diff
@@ -78,7 +78,17 @@ else if (result.getStatus() == UNRESOLVED && selector instanceof UniqueIdSelecto
 		}
 	}
 
-	static @Nullable TestSource toSource(DiscoverySelector selector) {
+	private static @Nullable TestSource toSourceSafely(DiscoverySelector selector) {
+		try {
+			return toSource(selector);
+		}
+		catch (Exception e) {
```

Comment:
I would expect a `rethrowIfUnrecoverable` here.

Reply:
Added in 090e4d98886df28cccf229525763fbbd6d45600a